blaze/expr/tests/test_arithmetic.py

```diff
@@ -72,3 +72,15 @@ def test_dir():
 def test_arith_ops_promote_dtype():
     r = w + z
     assert r.dshape == dshape('5 * 3 * float64')
+
+
+def test_str_arith():
+    assert isinstance(cs + cs, Add)
+    assert isinstance(cs * 1, Mult)
+    assert isinstance(cs % cs, Mod)
```

Comment:
I wonder if we should have separate string operations rather than overload `Add`.  The fact that these are the same in our minds is mostly a Python thing but may not carry over well to other backends.  It may be more clear to have a string concatenation operator.  Being explicit in this way might help when building backends because we won't have to check the dtype in `Add` operations for this case.